src/features/security-cloud-game/SecurityCloudGame.js

```diff
@@ -1,82 +1,115 @@
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
-import { FormattedMessage } from 'react-intl';
-import throttle from 'lodash/throttle';
-import Draggable from 'react-draggable';
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+import FocusTrap from '../../components/focus-trap';
 
 import DragCloud from './DragCloud';
 import DropCloud from './DropCloud';
 import messages from './messages';
+import { checkOverlap, getGridPosition, getRandomCloudPosition } from './utils';
 
 import './SecurityCloudGame.scss';
 
-class SecurityCloudGame extends Component {
-    static propTypes = {
-        /** Width/Height to set the drag and drop clouds to.  Defaults to 64 */
-        cloudSize: PropTypes.number,
-        /** Height to set the game to */
-        height: PropTypes.number.isRequired,
-        /** Function to call when the `DragCloud` is successfuly dropped onto the `DropCloud` */
-        onValidDrop: PropTypes.func,
-        /** Width to set the game to */
-        width: PropTypes.number.isRequired,
-    };
+// pick these numbers to balance accessibility and game complexity
+const CLOUD_SIZE_RATIO = 5;
+const GRID_TRACK_SIZE_RATIO = 20;
+
+const SecurityCloudGame = ({ height, intl, onValidDrop, width }) => {
+    const [cloudSize, setCloudSize] = useState(null);
+    const [gridTrackSize, setGridTrackSize] = useState(null);
+    const [dropCloudPosition, setDropCloudPosition] = useState(null);
+    const [dragCloudPosition, setDragCloudPosition] = useState(null);
+    const [gameBoardHeight, setGameBoardHeight] = useState(null);
```

Comment:
There are no good default values for these states, and we don't really need defaults either. The dependency chain looks like this:
`width, height <- gameBoardHeight <- cloudSize, gridTrackSize <- dragCloudPosition, dropCloudPosition`
Since DragCloud and DropCloud are only rendered if their corresponding position exists, by the time the initial render happens, all upstream properties would already be defined.  either. 

I combined `gameBoardHeight`, `cloudSize`, and `gridTrackSize` into one state object `layout` to avoid having to re-render SecurityCloudGame multiple times.